Support OL mapbox styles

gberaudo opened the pull request:
- [x] honor zIndex
- [x] investigate draw along line
- [x] investigate incorrect clipping
- [x] investigate decluttering (single layer)

gberaudo committed:
Fix the fix

gberaudo committed:
Try ol-mapbox-style example

gberaudo committed:
Reapply style and use geometry style if available

gberaudo commented:
![image](https://user-images.githubusercontent.com/7294662/134650793-9d15c686-b81e-4ac9-9218-314c542f8748.png)

gberaudo committed:
Show both demos

gberaudo committed:
Refactoring

gberaudo committed:
Use the feature-complete render API by default

OpenLayers has 2 ways to render features:
- the immediate API which is public and allows to simply render
  features on a canvas (but is uncomplete);
- the render API which is used by OL itself to render vector layers,
  as such it is feature-complete but was not formally intended for
  external use.

gberaudo committed:
Add prettier, eslint and github action

gberaudo committed:
Update action

gberaudo committed:
Fix tests

gberaudo committed:
Remove ts extension because of typescript whims

gberaudo committed:
Update dependencies

gberaudo committed:
Show node version

gberaudo committed:
Make tests pass

gberaudo committed:
Remove useless files from packaged npm

gberaudo committed:
Try to publish some version

gberaudo committed:
Add a license

gberaudo committed:
Skip lib

gberaudo committed:
Demo (broken) decluttering

gberaudo committed:
Fix run build

gberaudo committed:
Improve readme

gberaudo committed:
Hopefuly fix tests

gberaudo committed:
Fix rendering decluttered layer

gberaudo commented:
Fixes #3.

gberaudo committed:
Add some documentation

gberaudo committed:
Hardcode output dimensions

fredj commented on package.json:
```diff
@@ -1,43 +1,57 @@
-    "ts-jest": "^27.0.3",
-    "typescript": "^4.3.2",
-    "vite": "^2.3.8"
+    "lit": "2.0.0-rc.2",
```
lit 2.0.0 is out !

fredj commented on package.json:
```diff
@@ -1,43 +1,57 @@
+    "@babel/preset-typescript": "^7.15.0",
+    "@openlayers/eslint-plugin": "^4.0.0",
+    "@types/jest": "^27.0.2",
+    "@types/ol": "^6.5.3",
```
Not needed with recent version of ol (untested)

fredj reviewed (approved):
looks very good, thanks !

gberaudo committed:
Update dependencies

gberaudo committed:
Remove OL types dependency

3 benefits:
- one less dependency;
- not outdated anymore;
- navigate to source code in vscode instead of type.
